pkgs/applications/audio/mpg123/default.nix

```diff
@@ -1,31 +1,53 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchurl
 , makeWrapper
-, alsa-lib
+, pkg-config
 , perl
-, withConplay ? !stdenv.targetPlatform.isWindows
+, alsa-lib
+, withPulse ? stdenv.hostPlatform.isLinux
+, libpulseaudio
+, AudioUnit
+, AudioToolbox
+, withJack ? stdenv.hostPlatform.isUnix
+, jack
+, withConplay ? !stdenv.hostPlatform.isWindows
 }:
 
 stdenv.mkDerivation rec {
   pname = "mpg123";
-  version = "1.26.5";
+  version = "1.28.2";
 
   src = fetchurl {
     url = "mirror://sourceforge/${pname}/${pname}-${version}.tar.bz2";
-    sha256 = "sha256-UCqX4Nk1vn432YczgCHY8wG641wohPKoPVnEtSRm7wY=";
+    sha256 = "006v44nz4nkpgvxz1k2vbbrfpa2m47hyydscs0wf3iysiyvd9vvy";
   };
 
   outputs = [ "out" ] ++ lib.optionals withConplay [ "conplay" ];
 
-  nativeBuildInputs = lib.optionals withConplay [ makeWrapper ];
+  nativeBuildInputs = lib.optionals withConplay [ makeWrapper ]
+    ++ lib.optionals (withPulse || withJack) [ pkg-config ];
 
   buildInputs = lib.optionals withConplay [ perl ]
-    ++ lib.optionals (!stdenv.isDarwin && !stdenv.targetPlatform.isWindows) [ alsa-lib ];
+    ++ lib.optionals (stdenv.hostPlatform.isLinux) [ alsa-lib ]
+    ++ lib.optionals withPulse [ libpulseaudio ]
+    ++ lib.optionals (stdenv.hostPlatform.isDarwin) [ AudioUnit AudioToolbox ]
+    ++ lib.optionals withJack [ jack ];
 
-  configureFlags = lib.optional
-    (stdenv.hostPlatform ? mpg123)
+  configureFlags = [
+    "--with-audio=${lib.strings.concatStringsSep "," (
+      lib.optional withJack "jack"
+      ++ lib.optional withPulse "pulse"
+      ++ lib.optional stdenv.hostPlatform.isLinux "alsa"
+      ++ lib.optional stdenv.hostPlatform.isDarwin "coreaudio"
+      ++ [ "dummy" ]
+    )}"
```

Comment:
I don't like this but it's required to get a reasonable API preference order, otherwise we get `alsa oss jack pulse dummy` in 1.26.5 and `pulse alsa oss jack dummy` in 1.28.2 which are kinda sub-optimal?